aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -209,18 +210,28 @@ void initialize(
   @Override
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
     if (catalogProperties != null) {
-      Map<String, String> tableSpecificCatalogProperties =
-          ImmutableMap.<String, String>builder()
-              .putAll(catalogProperties)
-              .put(
-                  AwsProperties.LAKE_FORMATION_DB_NAME,
-                  IcebergToGlueConverter.getDatabaseName(
-                      tableIdentifier, awsProperties.glueCatalogSkipNameValidation()))
-              .put(
-                  AwsProperties.LAKE_FORMATION_TABLE_NAME,
-                  IcebergToGlueConverter.getTableName(
-                      tableIdentifier, awsProperties.glueCatalogSkipNameValidation()))
-              .build();
+      Map<String, String> tableSpecificCatalogProperties = Maps.newHashMap(catalogProperties);
+      tableSpecificCatalogProperties.put(
+          AwsProperties.LAKE_FORMATION_DB_NAME,
+          IcebergToGlueConverter.getDatabaseName(
+              tableIdentifier, awsProperties.glueCatalogSkipNameValidation()));
+      tableSpecificCatalogProperties.put(
+          AwsProperties.LAKE_FORMATION_TABLE_NAME,
+          IcebergToGlueConverter.getTableName(
+              tableIdentifier, awsProperties.glueCatalogSkipNameValidation()));
+      if (awsProperties.glueLakeFormationEnabled()) {
+        try {
+          glue.getTable(
+              GetTableRequest.builder()
+                  .databaseName(tableIdentifier.namespace().toString())
+                  .name(tableIdentifier.name())
+                  .build());
+          awsProperties.setS3PrefetchClientEnabled(true);
+        } catch (GlueException e) {
+          awsProperties.setS3PrefetchClientEnabled(false);
+          LOG.warn("Suppressing failure when getting Glue table", e);
```

Comment:
logging warning already means suppressing failure, no need to say that again. Maybe `"Fail to get Glue table information, cannot determine S3 prefetch strategy"`?

Also this will hit error very often, especially for new table creation. We should do debug log level.